src/TLuaInterpreter.cpp

```diff
@@ -5370,52 +5409,110 @@ int TLuaInterpreter::setStrikeOut(lua_State* L)
+    int s = 0;
+    if (lua_gettop(L) > 1) { // Have more than one argument so first must be a console name
+        if (!lua_isstring(L, ++s)) {
+            lua_pushfstring(L,
+                            "setUnderline: bad argument #%d type (window name, as string expected {omission selects \"main\" console window}, got %s!",
+                            s, luaL_typename(L, s));
+            return lua_error(L);
+        } else {
+            windowName = QString::fromUtf8(lua_tostring(L, s));
+        }
+    }
+
+    bool isAttributeEnabled;
+    if (!lua_isboolean(L, ++s)) {
+        lua_pushfstring(L, "setUnderline: bad argument #%d type (enable underline attribute as boolean expected, got %s!)", s, luaL_typename(L, s));
+        lua_error(L);
+        return 1;
+    } else {
+        isAttributeEnabled = lua_toboolean(L, s);
+    }
+
+    if (windowName.isEmpty() || windowName.compare(QStringLiteral("main"), Qt::CaseSensitive) == 0) {
+        host.mpConsole->setDisplayAttributes(TChar::Underline, isAttributeEnabled);
+        // Always succeeds on main console:
+        lua_pushboolean(L, true);
+        return 1;
+    } else {
+        if (mudlet::self()->setDisplayAttributes(&host, windowName, TChar::Underline, isAttributeEnabled)) {
+            lua_pushboolean(L, true);
+            return 1;
+        } else {
+            lua_pushnil(L);
+            lua_pushfstring(L, "window \"%s\" does not exist", windowName.toUtf8().constData());
+            return 2;
+        }
     }
-    return 0;
 }
 
 // Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#debugc -- not #debug - compare GlobalLua
 int TLuaInterpreter::debug(lua_State* L)
 {
     Host& host = getHostFromLua(L);
-    int nbargs = lua_gettop(L);
-    QString luaDebugText = "";
-    for (int i = 0; i < nbargs; i++) {
-        luaDebugText += (nbargs > 1 ? " (" + QString::number(i + 1) + ") " : " ") + lua_tostring(L, i + 1);
-        auto green = QColor(Qt::green);
-        auto blue = QColor(Qt::blue);
-        auto black = QColor(Qt::black);
-        QString s1 = QString("[DEBUG:]");
-        QString s2 = QString("%1\n").arg(luaDebugText);
-        if (host.mpEditorDialog) {
-            host.mpEditorDialog->mpErrorConsole->printDebug(blue, black, s1);
-            host.mpEditorDialog->mpErrorConsole->printDebug(green, black, s2);
+    int n = lua_gettop(L);
+    if (!n) {
+        // Nothing to show
+        return 0;
+    }
+
+    QString luaDebugText;
+    if (n > 1) {
+        for (int i = 0; i < n; ++i) {
+            luaDebugText += QStringLiteral(" (%1) %2").arg(QString::number(i + 1), QString::fromUtf8(lua_tostring(L, i + 1)));
         }
+    } else {
+        // n == 1
+        luaDebugText = QStringLiteral(" %1").arg(QString::fromUtf8(lua_tostring(L, 1)));
     }
+    luaDebugText.append(QChar::LineFeed);
+
+    if (host.mpEditorDialog) {
+        host.mpEditorDialog->mpErrorConsole->print(QLatin1String("[DEBUG:]"), QColor(Qt::white), QColor(Qt::blue));
```

Comment:
Not sure why I did that when I originally created the code - it might have been that blue on black is harder to read (less fg-bg contrast) than blue on white - or it might have been trying to produce a unique colour combination for explicit usage of the lua `debug` function.

Reply:
I think it should still the same, the old combination works well

Reply:
Okay, I'll restore the original colours.

Reply:
This has been done at some point.